### PR TITLE
Implement !latex

### PR DESCRIPTION
**Pull request by TRManderson**

### CODE REVIEW

File: test/test_latex.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""
+Tests for uqcsbot.scripts.cat
+"""
+from .conftest import MockUQCSBot
+from .helpers import generate_message_object
+
+
+def test_latex(uqcsbot: MockUQCSBot):
```
Comment (mitchmcdee) [inline test/test_latex.py]: this test needs updating to the new test framework (also, can you adopt the new styling too)

File: uqcsbot/scripts/latex.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                "id": 1,
+            }
+        ],
+        unfurl_links=False,
+        unfurl_media=False,
+    )
+
+
+@bot.on_command("latex")
+def handle_latex_cmd(cmd: Command):
```
Comment (mitchmcdee) [inline uqcsbot/scripts/latex.py]: nit: prefer `command` over `cmd` to align with other scripts

File: uqcsbot/scripts/latex.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    """
+    `!latex CONTENT` - Renders `CONTENT` to LaTeX and sends it to Slack.
+    `$$ CONTENT $$` also works.
+    """
+    if not cmd.has_arg():
+        bot.post_message(cmd.channel, "No data provided")
```
Comment (mitchmcdee) [inline uqcsbot/scripts/latex.py]: needs to be updated to `cmd.channel_id` here and below

File: uqcsbot/scripts/latex.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    url = f"http://latex.codecogs.com/gif.latex?{quote(data)}"
+    bot.post_message(
+        channel, 
+        text=f"LaTeX render for \"{data}\"",
+        attachments=[
+            {
+                "fallback": f"Codecogs LaTeX image for {data}",
+                "image_url": url,
+                "from_url": url,
+                "id": 1,
```
Comment (mitchmcdee) [inline uqcsbot/scripts/latex.py]: what's `id` used for? (is it required?)

Comment (TRManderson) [reply uqcsbot/scripts/latex.py]: No idea, I think it's just for ordering? I copied it from a deconstructed `message_changed` event.

Comment (mitchmcdee) [reply uqcsbot/scripts/latex.py]: strip out as much as possible imo, if it ain't needed turf it

File: test/conftest.py [hunk 5]
```diff
@@ -166,15 +166,15 @@ def mocked_chat_postMessage(self, **kwargs):
...
         channel = self.channels.get(channel_id_or_name)
         if channel is None:
             return {'ok': False}
 
-        message = {'text': text, 'ts': str(time.time())}
+        message = {'ts': str(time.time()), **kwargs}
         self.test_messages[channel.id].append(message)
-        message_event = {'type': 'message', 'channel': channel.id, **message}
+        message_event = {'type': 'message', **message}
+        message_event['channel'] = channel.id
```
Comment (mitchmcdee) [inline test/conftest.py]: why is this brought outside?

File: test/test_latex.py [hunk 6]
```diff
@@ -1,12 +1,13 @@
 """
 Tests for uqcsbot.scripts.cat
 """
```
Comment (mitchmcdee) [inline test/test_latex.py]: can you move this to be a function doc instead like the other tests? (and update it from being for !cat)

File: test/test_latex.py [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
 def test_latex(uqcsbot: MockUQCSBot):
-    message = generate_message_object("!latex test_memes")
-    uqcsbot.test_handle_event(message)
-    assert len(uqcsbot.test_posted_messages) == 1
-    assert uqcsbot.test_posted_messages[-1].text == "LaTeX render for \"test_memes\""
\ No newline at end of file
+    uqcsbot.post_message(TEST_CHANNEL_ID, "!latex test_memes")
+    messages = uqcsbot.test_messages.get(TEST_CHANNEL_ID, [])
+    assert len(messages) == 2
+    print(messages)
```
Comment (mitchmcdee) [inline test/test_latex.py]: remove print

File: test/test_latex.py [hunk 8]
```diff
@@ -1,12 +1,13 @@
...
+    assert messages[-1]["text"] == "LaTeX render for \"test_memes\""
+    assert len(messages[-1].get("attachments", [])) == 1
```
Comment (mitchmcdee) [inline test/test_latex.py]: new line

Comment (mitchmcdee) [reply test/test_latex.py]: also why not something like `messages[-1].get('attachments') is not None`?

File: test/conftest.py (hunk 5 above)
Comment (TRManderson) [reply test/conftest.py]: The message event is what the bot would receive, which I'm pretty sure is always a channel id. The splat was overwriting it with whatever the arg passed in was.

File: test/test_latex.py (hunk 8 above)
Comment (TRManderson) [reply test/test_latex.py]: I want it to return exactly 1 attachment

**Comment by mitchmcdee** [comment]
Tested and ready to merge?

**Comment by TRManderson** [comment]
Yep! `pytest` and real-world tests.

**Pull request merged**